### ISSUE TITLE
[P1] Architecture upgrade — replace channel-wise bias with cross-attention text injection

### ISSUE THREAD

**Issue by AcarBasaran** [ahmetnecirvandogan/sketch-generator#45]
**Priority:** P1 (architectural improvement; not blocking first run, but blocking final paper-quality model)
**Depends on:** #20 (model scaffold) ✓
**Blocks:** none directly; will improve text fidelity meaningfully

## Why this matters

Right now, our model uses the **simplest possible** way to feed text into the U-Net.

```python
# pbr_model/model.py, line ~265
text_feat = self.text_proj(text_feat)        # (B, 16c)
x4 = x4 + text_feat[:, :, None, None]        # broadcast across spatial
```

This means:
- The text vector is added **once**, at the deepest U-Net layer (the bottleneck)
- It is added as a **channel-wise bias** — the same shift to every spatial position
- The text cannot say "put red color here, blue color there" — only "shift all channels uniformly across the whole bottleneck"

**Effect:** the network can learn global text → image associations ("silk → glossy overall") but it cannot learn spatial associations ("houndstooth → black-and-white squares specifically in the cloth body region").

Every modern text-to-image model uses **cross-attention** instead (Stable Diffusion, ControlNet, DALL-E, Imagen). Cross-attention lets each spatial position decide which parts of the text feature it should pay attention to. This is much more expressive.

## What to change

Replace the simple channel-wise bias with **cross-attention layers** at multiple decoder stages.

### How cross-attention works (short version)

For each pixel in the U-Net feature map:
1. Compute a query vector from the pixel
2. Compute key + value vectors from the text feature (use the full sequence from CLIP, not the pooled output)
3. Pixel attends to all text tokens, weighted by query · key similarity
4. Weighted sum of values is added back to the pixel feature

This means each pixel can ask "which words in the prompt are relevant to me?" and use them.

### Where to inject

Add cross-attention blocks at the bottleneck and at each decoder up-stage:

```
encoder    →    bottleneck (cross-attn here) ──┐
                                                ▼
        ──    decoder up1 (cross-attn here)
                                                ▼
        ──    decoder up2 (cross-attn here)
                                                ▼
        ──    decoder up3 (cross-attn here)
                                                ▼
        ──    decoder up4
                                                ▼
                                              outputs
```

This matches Stable Diffusion's pattern.

### What CLIP gives us

Right now we use CLIP's `pooler_output` (one 512-dim vector per prompt). For cross-attention we want the **full token sequence** — `last_hidden_state` from CLIP, shape `(B, sequence_length, 512)`. Update `CLIPTextEncoder` to return both:

```python
class CLIPTextEncoder:
    def forward(self, prompts):
        out = self.text_model(...)
        return {
            "pooled":   out.pooler_output,        # (B, 512)
            "sequence": out.last_hidden_state,    # (B, T, 512) for cross-attention
        }
```

Cross-attention layers consume `sequence`. The old bottleneck-bias path (using `pooled`) can be kept for backward compatibility.

## Files to change

- `pbr_model/model.py` — add `CrossAttentionBlock` class; modify `PBRModel.forward` to apply cross-attention at bottleneck + 3 decoder up-stages; update `TextEncoder` and `CLIPTextEncoder` to return both pooled and sequence

## How to test

- Forward pass on (B=2, sketch=(2,3,512,512), prompt=list[str] of len 2) returns expected shapes for both Variant A and Variant B
- Backward pass: gradients reach U-Net + cross-attention blocks; CLIP weights are still frozen (`requires_grad=False`)
- 100-step smoke training on the existing `dataset/` runs without NaN; loss decreases at least 10% from step 1 to step 100

## Estimate

~200 lines. ~1 day. Mostly the cross-attention block class + threading the sequence through the U-Net.

## Notes

- Use multi-head attention (e.g. 8 heads). Standard PyTorch `nn.MultiheadAttention` works fine.
- Skip-connect cross-attention output to the U-Net feature: `feature = feature + attention_output` so we do not break existing learning.
- Initialize cross-attention weights to small values so the model behavior at step 0 is similar to without cross-attention. This makes training stable.
